datadog_checks_dev/datadog_checks/dev/tooling/license_headers.py

```diff
@@ -54,24 +57,28 @@ def validate_license_headers(
       - tests/docker
       - tests/compose
     """
-    root = check_path
     default_ignore = {pathlib.Path("tests/docker"), pathlib.Path("tests/compose")}
     ignoreset = frozenset(default_ignore.union(extra_ignore or set()))
 
-    def walk_recursively(path):
+    def walk_recursively(path, gitignore_matcher):
```

Comment:
I'm not sure if it should be done in this PR or the base one, but we also should ignore vendored dependencies, for example `php_fpm/datadog_checks/php_fpm/vendor` and `tokumx/datadog_checks/tokumx/vendor/`

Reply:
I intended to add those (since it only happens twice in the whole repo, so they're special cases at the moment) here as needed:
https://github.com/DataDog/integrations-core/blob/664d3cae5065478b4624e93a106c493e5380ed52/datadog_checks_dev/datadog_checks/dev/tooling/commands/validate/license_headers.py#L15

I'd rather avoid (or postpone) overloading the core validation logic with all the little rules and exceptions that we may find. In fact, if anything, I think we could even remove the `default_ignore` above and have that the caller be fully responsible for setting all exceptions.

Reply:
I've now removed all hardcoded paths from the logic, and all paths to are passed explicitly by the caller. I've done this on #13417 though.

Reply:
Nice, thanks!

Reply:
I intended to leave that for a future PR where I intend to fix all licensing issues that the command catches. It would be awkward to change that in this PR as it's unrelated.

Reply:
#13511 will add that and fix all licensing issues to try to start from a clean slate.